### PR TITLE
[5888] Export SearchUI component so that it can easily be used by plugins

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5888

### CODE REVIEW

File: ui/app/src/components/Search/Search.tsx [hunk 1]
```diff
@@ -26,18 +26,22 @@ import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
...
 export function Search(props: SearchProps) {
-  const { mode = 'default', onSelect, embedded = false, onAcceptSelection, onClose } = props;
+  const { mode = 'default', onSelect, embedded = false, onAcceptSelection, onClose, parameters } = props;
 
   // region State
   const [filters, setFilters] = useSpreadState({
-    sortBy: undefined,
-    sortOrder: undefined
+    sortBy: parameters?.sortBy ?? undefined,
+    sortOrder: parameters?.sortOrder ?? undefined
```
Comment (rart) [inline ui/app/src/components/Search/Search.tsx]: I do wonder if this is even necessary. If these props aren't supplied, they will already be undefined. If only the sub-props are not supplied, they will also be undefined.
 ```suggestion
    sortBy: parameters?.sortBy ?? UNDEFINED,
    sortOrder: parameters?.sortOrder ?? UNDEFINED
```

File: ui/app/src/components/Search/utils.ts [hunk 2]
```diff
@@ -72,9 +72,14 @@ export interface URLDrivenSearchProps {
...
 }
 
+export interface SearchParameters extends Partial<ElasticParams> {
+  path?: string;
+}
+
 export interface SearchProps {
   mode?: 'default' | 'select';
   embedded?: boolean;
+  parameters?: SearchParameters;
```
Comment (rart) [inline ui/app/src/components/Search/utils.ts]:
```suggestion
  initialFilters?: SearchParameters;
```

File: ui/app/src/components/Search/utils.ts [hunk 3]
```diff
@@ -72,9 +72,14 @@ export interface URLDrivenSearchProps {
   onAcceptSelection?(items: DetailedItem[]): any;
 }
 
+export interface SearchParameters extends Partial<ElasticParams> {
```
Comment (rart) [inline ui/app/src/components/Search/utils.ts:75]:
```suggestion
export interface SearchFilters extends Partial<ElasticParams> {
```

File: ui/app/src/components/Search/Search.tsx [hunk 4]
```diff
@@ -70,6 +74,23 @@ export function Search(props: SearchProps) {
...
+      const newCheckedFilters = {};
+      Object.entries(parameters.filters).forEach(([key, filter]: [string, any]) => {
+        if (Array.isArray(filter)) {
+          newCheckedFilters[key] = filter.reduce((checked, key) => ({ ...checked, [key]: true }), {});
+        } else if (filter.date) {
+          newCheckedFilters[key] = `${filter.min ?? ''}TODATE${filter.max ?? ''}ID${filter.id}`;
+        } else {
+          newCheckedFilters[key] = `${filter.min ?? ''}TO${filter.max ?? ''}`;
+        }
+      });
```
Comment (rart) [inline ui/app/src/components/Search/Search.tsx]: Preferably this should be in a single place. Where does this concatenation and such usually occurs? I would have thought this all should be on the useSearchState hook.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/Search/utils.ts (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/Search/utils.ts]: per conversation, it'll be called 'initialParameters'

File: ui/app/src/components/Search/utils.ts (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/Search/utils.ts:75]: per conversation, it'll remain called 'SearchParameters'

**Review by rart — APPROVED** [review]

**Pull request merged**